backend/Controllers/ChatsController.cs

```diff
@@ -0,0 +1,93 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using AutoMapper;
+using backend.DTO.ChatControllerDTO;
+using backend.Models;
+using backend.DTO.UserControllerDTO;
+using AutoMapper.QueryableExtensions;
+
+
+namespace backend.Controllers;
+
+// GET api/chat/{id} -> { participant: {}, messages: [] }
+// GET api/chat/{id}/messages -> [MessageDTO]
+// GET api/chat/{id}/open -> Opens new SignalR connection
+// POST api/chat -> Create a new chat with user specified in the CreateChatDTO
+
+[ApiController]
+[Route("api/[controller]")]
+public class ChatsController : ControllerBase
+{
+    private readonly ILogger<ChatsController> _logger;
+    private readonly ChatDbContext _chatDbContext;
+    private readonly IMapper _mapper;
+
+    public ChatsController(ILogger<ChatsController> logger, ChatDbContext chatDbContext, IMapper mapper)
+    {
+        _logger = logger;
+        _chatDbContext = chatDbContext;
+        _mapper = mapper;
+    }
+
+    [HttpGet("{id}")]
+    [ProducesResponseType(typeof(GetChatDTO), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(int), StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> ChatById([FromRoute] int id, CancellationToken cancellationToken)
+    {
+        if (!Request.Cookies.ContainsKey("currentUserId"))
+        {
+            return BadRequest("currentUserId not present in the cookies collection.");
+        }
+
+        if (!int.TryParse(Request.Cookies["currentUserId"], out var currentUserId))
+        {
+            return BadRequest("currentUserId is present in the cookies collection, but has incorrect format.");
+        }
+
+        var chat = await _chatDbContext.Chats.FindAsync(id, cancellationToken);
+        if (chat is null)
+        {
+            return NotFound(id);
+        }
+
+        if (chat.UserId1 != currentUserId && chat.UserId2 != currentUserId)
+        {
+            return Forbid();
+        }
+
+        int participantId = chat.UserId2 == currentUserId ? chat.UserId2 : chat.UserId1;
```

Comment:
Should invert the ternary operator result – when currentUserId == UserId2, then participantId = UserId1, and vice versa